File: Source/CSharp/Expressive/Expressive/Operators/Logic/AndOperator.cs
using Expressive.Expressions;
using Expressive.Expressions.Binary.Logic;

namespace Expressive.Operators.Logic
{
    internal class AndOperator : OperatorBase
    {
        #region OperatorBase Members

        public override string[] Tags => new[] { "&&", "and" };

        public override IExpression BuildExpression(Token previousToken, IExpression[] expressions, ExpressiveOptions options)
        {
            return new AndExpression(expressions[0], expressions[1], options);
        }

        public override OperatorPrecedence GetPrecedence(Token previousToken)
        {
            return OperatorPrecedence.And;
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/operators/logic/AndOperator.java
package com.bijington.expressive.operators.logic;

import com.bijington.expressive.expressions.BinaryExpression;
import com.bijington.expressive.expressions.BinaryExpressionType;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.operators.OperatorBase;
import com.bijington.expressive.operators.OperatorPrecedence;

/**
 * Created by shaun on 11/07/2016.
 */
public class AndOperator extends OperatorBase {
    @Override
    public String[] getTags() { return new String[] { "&&", "and" }; }

    @Override
    public IExpression buildExpression(String previousToken, IExpression[] expressions) {
        return new BinaryExpression(BinaryExpressionType.And, expressions[0], expressions[1]);
    }

    @Override
    public int getPrecedence(String previousToken) {
        return OperatorPrecedence.And;
    }
}
